Frameworks/CoreGraphics/CGContext.mm

```diff
@@ -1636,22 +1668,141 @@ void CGContextSetCMYKFillColor(CGContextRef context, CGFloat cyan, CGFloat magen
+    size_t bitsPerComponent = 8;
+    size_t bytesPerRow = 4 * tileSize.size.width;
+    CGBitmapInfo bitmapInfo = kCGImageAlphaPremultipliedFirst | kCGBitmapByteOrder32Big;
+
+    woc::unique_cf<CGContextRef> patternContext{ CGBitmapContextCreate(
+        nullptr, tileSize.size.width, tileSize.size.height, bitsPerComponent, bytesPerRow, colorspace.get(), bitmapInfo) };
+    RETURN_HR_IF_NULL(E_UNEXPECTED, patternContext);
+
+    CGContextSetFillColorSpace(patternContext.get(), context->fillColorSpace());
+    // set the fill color
+    CGContextSetFillColor(patternContext.get(), components);
+
+    // Now we ask the user to draw
+    _CGPaternIssueCallBack(patternContext.get(), pattern);
+
+    // Get the image out of it
+    woc::unique_cf<CGImageRef> tileImage{ __CGContextCreateRenderableImage(CGBitmapContextCreateImage(patternContext.get())) };
+    RETURN_HR_IF_NULL(E_UNEXPECTED, tileImage);
+
+    ComPtr<ID2D1Bitmap> d2dBitmap;
+    RETURN_IF_FAILED(__CreateD2DBitmapFromCGImage(context, tileImage.get(), &d2dBitmap));
+
+    // Scale it by the inverted transform
+    // TODO: We have an issue with rotation, the CTM rotation should not affect the brush
+    CGSize size = CGSizeApplyAffineTransform(tileSize.size, CGAffineTransformInvert(context->CurrentGState().transform));
+
+    CGAffineTransform transform = __BitmapBrushTransformation(context,
+                                                              { { 0, 0 }, size.width, size.height },
+                                                              d2dBitmap->GetPixelSize(),
+                                                              _CGPatternGetTransformation(pattern));
+
+    ComPtr<ID2D1BitmapBrush1> bitmapBrush;
+    ComPtr<ID2D1DeviceContext> deviceContext = context->DeviceContext();
+    RETURN_IF_FAILED(deviceContext->CreateBitmapBrush(d2dBitmap.Get(),
+                                                      D2D1::BitmapBrushProperties1(D2D1_EXTEND_MODE_WRAP,
+                                                                                   D2D1_EXTEND_MODE_WRAP,
+                                                                                   context->CurrentGState().GetInterpolationModeForCGImage(
+                                                                                       tileImage.get())),
+                                                      D2D1::BrushProperties(1 /* the global alpha will be set during the draw sequence*/,
+                                                                            __CGAffineTransformToD2D_F(transform)),
+                                                      &bitmapBrush));
+
+    *brush = bitmapBrush.Detach();
+    return S_OK;
+}
+
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
-void CGContextSetStrokePattern(CGContextRef context, CGPatternRef pattern, const CGFloat* components) {
+void CGContextSetFillPattern(CGContextRef context, CGPatternRef pattern, const CGFloat* components) {
```

Comment:
>CGContextSetFillPattern [](start = 5, length = 23)

actually need to add Caveat and notes saying components are treated as RGBA for now.